File: jenkins-plugin/src/main/java/org/eazyportal/plugin/release/jenkins/action/FinalizeReleaseVersionActionFactory.java
package org.eazyportal.plugin.release.jenkins.action;

import hudson.Extension;
import hudson.model.InvisibleAction;
import org.eazyportal.plugin.release.core.action.FinalizeReleaseVersionAction;
import org.eazyportal.plugin.release.core.scm.ScmActions;

import java.io.Serializable;

@Extension
public class FinalizeReleaseVersionActionFactory extends InvisibleAction implements Serializable {

    public FinalizeReleaseVersionAction create(ScmActions scmActions) {
        return new FinalizeReleaseVersionAction(scmActions);
    }

}

File: gradle-plugin/src/main/kotlin/org/eazyportal/plugin/release/gradle/action/FinalizeReleaseVersionActionFactory.kt
package org.eazyportal.plugin.release.gradle.action

import org.eazyportal.plugin.release.core.action.FinalizeReleaseVersionAction
import org.eazyportal.plugin.release.gradle.model.EazyReleasePluginExtension

class FinalizeReleaseVersionActionFactory : ReleaseActionFactory<FinalizeReleaseVersionAction> {

    override fun create(extension: EazyReleasePluginExtension): FinalizeReleaseVersionAction =
        FinalizeReleaseVersionAction(extension.projectActionsFactory, extension.scmActions)

}
